news-explorer/src/components/Main/Main.js

```diff
@@ -1,15 +1,64 @@
 import SearchForm from "../SerchForm/SerchForm";
 import NewsCardList from "../NewsCardList/NewsCardList";
 import About from "../About/About";
+import getNews from '../../utils/NewsApi';
+import Preloader from "../Preloader/Preloader";
+import React, { useState } from 'react';
 
 
-function Main() {
+
+
+function Main(props) {
+
+    const [renderPreloader, setRenderPreloader] = React.useState(false);
+    const [renderNotFound, setRenderNotFound] = React.useState(false);
+    const [cardsList, setCardsList] = React.useState([]);
+
+    function handelFormSubmit(keyword) {
+        setRenderNotFound(false);
+        props.setRenderCards(false);
+        setRenderPreloader(true);
+        getNews(keyword)
+            .then((data) => {
+                if (data.status !== 'ok') {
+                    setRenderNotFound(true);
+                } else {
+                    console.log(data);
+                    setCardsList(data.articles);
+                    localStorage.setItem('newsArticles', JSON.stringify(data.articles));
+                    localStorage.setItem('keyword', keyword);
+                    setRenderPreloader(false);
+                    props.setRenderCards(true);
+                }
+            })
```

Comment:
You need to `catch` possible errors at the very end of any server request.

Please, correct it everywhere